.github/workflows/ci.yml

```diff
@@ -19,10 +19,10 @@ jobs:
         -   uses: actions/checkout@v4
             with:
                 submodules: true
+        -   uses: astral-sh/setup-uv@v5
```

Comment:
Any reason not to use [ruff-action](https://github.com/astral-sh/ruff-action)? I don't recall why we went with installing it manually when we switched..

Reply:
The whole point of this PR is to encode ruff version ranges in `pyproject.toml`. AFAICT, `ruff-action` ignores that?

Reply:
I don't know if that was the case in the past, but their documentation says they do read `pyproject.toml` to get the version: https://github.com/astral-sh/ruff-action?tab=readme-ov-file#install-specific-versions